### PR TITLE
Migrate S2053: Scaffold the rule for the new SE

### PR DESCRIPTION
**Pull request by zsolt-kolbay-sonarsource**
Part of #7558

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/HashesShouldHaveUnpredictableSaltTest.cs [hunk 1]
```diff
@@ -20,70 +20,130 @@
...
+        var context2 = new HashesShouldHaveUnpredictableSalt.LocationContext(location, "2nd message");
+        var context3 = new HashesShouldHaveUnpredictableSalt.LocationContext(null, "message");
+        var context4 = new HashesShouldHaveUnpredictableSalt.LocationContext(location, "message");
+
+        context1.Equals(null).Should().BeFalse();
+        context1.Equals(new object()).Should().BeFalse();
+        context1.Equals(context2).Should().BeFalse();
+        context1.Equals(context3).Should().BeFalse();
+        context1.Equals(context1).Should().BeTrue();
+        context1.Equals(context4).Should().BeTrue();
```
Comment (mary-georgiou-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/HashesShouldHaveUnpredictableSaltTest.cs:147]: I don't understand fully the purpose of this test. :/ 
Why is is needed and is it also needed for the migrated rule?

**Review by mary-georgiou-sonarsource — CHANGES_REQUESTED** [review]
Polishing

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/HashesShouldHaveUnpredictableSaltTest.cs (hunk 1 above)
Comment (zsolt-kolbay-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/HashesShouldHaveUnpredictableSaltTest.cs:137]: `LocationContext` is a nested class inside the Sonar version of `HashesShouldHaveUnpredictableSalt`. This UT just validates the implemented `Equals` method inside it, to test that it works correctly and to have code coverage.
I don't plan to use such a type in the Roslyn implementation for now, so I'm not making a copy of this for the new rule implementation.

Comment (mary-georgiou-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/HashesShouldHaveUnpredictableSaltTest.cs:137]: great thanks!

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGTM

File: analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/HashesShouldHaveUnpredictableSaltBase.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
+
+public abstract class HashesShouldHaveUnpredictableSaltBase : SymbolicRuleCheck
+{
+    protected const string DiagnosticId = "S2053";
+    protected const string MessageFormat = "{0}";
+    protected const string MakeSaltUnpredictableMessage = "Make this salt unpredictable.";
+    protected const string MakeThisSaltLongerMessage = "Make this salt at least 16 bytes.";
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/HashesShouldHaveUnpredictableSaltBase.cs:28]: For these two, `private` should be enough - in the next PR
```suggestion
    private const string MakeSaltUnpredictableMessage = "Make this salt unpredictable.";
    private const string MakeThisSaltLongerMessage = "Make this salt at least 16 bytes.";
```

**Review by pavel-mikula-sonarsource — APPROVED** [review]
Low coverage is expected at this stage

**Pull request merged**